Blish HUD/GameServices/ArcDpsService.cs

```diff
@@ -1,96 +1,122 @@
+#if DEBUG
+        public static long Counter = 0;
+#endif
+
+        private SocketListener _server;
+
+        private Stopwatch _stopwatch;
+
+        /// <summary>
+        ///     Indicates if arcdps updated <see cref="HudIsActive" /> in the last second (it should every in-game frame)
+        /// </summary>
+        public bool RenderPresent { get; private set; }
+
+        /// <summary>
+        ///     Indicates if arcdps currently draws its HUD (not in character select, cut scenes or loading screens)
+        /// </summary>
+        public bool HudIsActive
+        {
+            get
+            {
+                lock (WatchLock)
+                {
+                    return _hudIsActive;
+                }
+            }
+            private set
+            {
+                lock (WatchLock)
+                {
+                    _stopwatch.Restart();
+                    _hudIsActive = value;
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Holds unprocessed combat data
+        /// </summary>
+        public event RawCombatEvent OnRawCombatEvent;
 
         protected override void Initialize()
         {
-            Server = new NamedPipeServerStream("BHUDrender");
-            Mut = new Mutex(false);
+            _stopwatch = new Stopwatch();
+            _server = new SocketListener(10, 200_000);
+            _server.ReceivedMessage += MessageHandler;
+#if DEBUG
+            OnRawCombatEvent += data =>
+            {
+                Interlocked.Increment(ref Counter);
+            };
+#endif
         }
 
         protected override void Load()
         {
-            Cancel = new CancellationToken(false);
-            Watch = Stopwatch.StartNew();
-            Task.Run(() =>
-            {
-                while (true)
-                    try
-                    {
-                        Server.WaitForConnection();
-                        var obj = Serializer.Deserialize<Arc>(Server);
-                        Server.Disconnect();
+            _server.Start(new IPEndPoint(IPAddress.Loopback, 8214));
+            _stopwatch.Start();
+        }
 
-                        if (obj.Msgtype == Mtype.Imgui)
-                        {
-                            Mut.WaitOne();
-                            try
-                            {
-                                Watch.Restart();
-                            }
-                            finally
-                            {
-                                Mut.ReleaseMutex();
-                            }
-                        }
-                        else
+        protected override void Unload()
+        {
+            _stopwatch.Stop();
+            _server.Stop();
+            RenderPresent = false;
+        }
+
+        protected override void Update(GameTime gameTime)
+        {
+            TimeSpan elapsed;
+            lock (WatchLock)
+            {
+                elapsed = _stopwatch.Elapsed;
```

Comment:
Could you elaborate on why you're using a stopwatch instead of the elapsed time provided by `gameTime`?

Reply:
because i cannot access the elapsed time on the other thread. and if i wanted to use it, then i would have to set some bool "resetclock" or something and do it inside the update then. if you want that, i can change it to that way

Reply:
Ahh, well that makes sense.  Thanks for the clarification!